### PR TITLE
Switch to GitHub actions for CI

### LINKED ISSUE #220: Investigate alternative CI solutions
Travis takes so long for android tests, would like to explore this more.

### PR DESCRIPTION
**Pull request by ZacSweers**
Resolves #220, porting mostly from RxDogTag and Crumb configurations

A little WIP as I try out the emulator solution (http://vgaidarji.me/blog/2019/01/27/github-actions/).

### CODE REVIEW

**Comment by ZacSweers** [comment]
The emulator support on actions is missing some documentation and usage instructions. Asked for more details here: https://github.community/t5/GitHub-Actions/How-to-use-the-Android-emulator-from-GitHub-Actions/m-p/33121/highlight/true#M1374

**Comment by ZacSweers** [comment]
In a worst case scenario - may just keep travis CI _just_ for instrumentation tests and run everything else on GitHub. This way Travis would only be running a subset of the tests and builds.

For later comparisons - right now a full travis build takes ~18min

**Comment by ZacSweers** [comment]
Wowsers, seems most of the travis CI build is just downloading the android SDK, so removing other tests just saves 2min 😞

**Comment by ZacSweers** [comment]
Sigh, even the workaround doesn't work. Emulator never shows up

**Comment by ZacSweers** [comment]
Nice wins!

**Comment by ZacSweers** [comment]
image

**Comment by ZacSweers** [comment]
@ShaishavGandhi this is ready for another pass

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.0-all.zip
```
Comment (ShaishavGandhi) [inline gradle/wrapper/gradle-wrapper.properties]: Nice

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+name: CI
+
+on: [push]
```
Comment (ShaishavGandhi) [inline .github/workflows/ci.yml]: Add `pr` to it? When people with forks make a PR, it won't trigger the `push` job since it's not a push on this repo

Comment (ZacSweers) [reply .github/workflows/ci.yml]: Are you sure about that? We use this pattern with the other repos, kotlinpoet uses this too

Comment (ShaishavGandhi) [reply .github/workflows/ci.yml]: I think so. I saw this in another repo. We can merge as is and i can test in a followup.

**Pull request merged**